test/dataframe.jl

```diff
@@ -332,43 +332,47 @@ module TestDataFrame
-                                    max = [4.0, 3.0, nothing, nothing, nothing, nothing],
+                                    max = [4.0, 3.0, "d", "c", Date(2004), nothing],
+                                    nunique = [nothing, nothing, 4, 3, 4, 2],
                                     nmissing = [nothing, 1, nothing, 1, nothing, nothing],
                                     eltype = [Int, Int, String, String, Date, eltype(df[:catarray])])
         describe_output_all_stats = DataFrame(variable = [:number, :number_missing, 
-                                                          :non_number, :non_number_missing,
+                                                          :string, :string_missing,
                                                           :dates, :catarray],
                                               mean = [2.5, 2.0, nothing, nothing, nothing, nothing],
                                               std = [Compat.std(df[:number]), 1.0, nothing, 
                                                      nothing, nothing, nothing],
-                                              min = [1.0, 1.0, nothing, nothing, nothing, nothing],
+                                              min = [1.0, 1.0, "a", "a", Date(2000), nothing],
                                               q25 = [1.75, 1.5, nothing, nothing, nothing, nothing],
                                               median = [2.5, 2.0, nothing, nothing, nothing, nothing],
                                               q75 = [3.25, 2.5, nothing, nothing, nothing, nothing],
-                                              max = [4.0, 3.0, nothing, nothing, nothing, nothing],
-                                              nunique = [nothing, nothing, 4, 4, 4, 2],
+                                              max = [4.0, 3.0, "d", "c", Date(2004), nothing],
+                                              nunique = [nothing, nothing, 4, 3, 4, 2],
                                               nmissing = [nothing, 1, nothing, 1, nothing, nothing],
+                                              first = [1, 1, "a", "a", Date(2000), 1],
+                                              last = [4, missing, "d", missing, Date(2004), 2],
                                               eltype = [Int, Int, String, String, Date, 
                                                         eltype(df[:catarray])])
 
         
         # Test that it works as a whole, without keyword arguments
         @test describe_output == describe(df)
 
+        # Test that it works with one stats argument
+        @test describe_output[[:variable, :mean]] == describe(df, stats = [:mean])
+
         # Test that it works with all keyword arguments
-        @test describe_output_all_stats == 
-              describe(df, stats = [:mean, :std, :min, :q25, :median, :q75, :max, 
-                       :nunique, :nmissing, :eltype])
+        @test describe_output_all_stats ≅ describe(df, stats = [:all])
```

Comment:
Looking at this, it just occurred to me that it would be nice to allow passing `:all` directly. Actually, it would make even more sense to reject `:all` when it's inside an array, since it's redundant with other entries. So we could just check that if `stats` is a `Symbol` it's equal to `:all`.

Reply:
Made this change and added it to the docs and errors.